src/components/Modal/Modal.tsx

```diff
@@ -1,115 +1,58 @@
 import * as React from 'react';
 import { themr, ThemedComponentClass } from 'react-css-themr';
+import { classNames } from '@shopify/react-utilities/styles';
+
 import { Keys } from '../../types';
 import { MODAL } from '../ThemeIdentifiers';
 import KeypressListener from '../KeypressListener';
-import helpers from './helpers';
 import Dialog from './Dialog';
-import * as baseTheme from './Modal.scss';
-// import * as velocity from 'velocity-animate';
-
-const velocity = require('velocity-animate');
-
-const bodyStyle = (pading: string, overflow: string): void => {
-  const body = document.getElementsByTagName('body');
-  body[0].style.paddingRight = pading;
-  body[0].style.overflow = overflow;
-};
-
-const getModalElement = (id: string) => {
-  return {
-    modal: helpers.getElement(`modal-${id}`),
-    dialog: helpers.getElement(`dialog-${id}`),
-  };
-};
 
-export type Size = 'Small' | 'Medium' | 'Large' | number;
-
-export interface IAnimate {
-  in(modal?: Element, dialog?: Element): void;
-  out(modal?: Element, dialog?: Element): void;
-}
+import * as baseTheme from './Modal.scss';
 
-export interface ITrigger {
-  body: string;
-  animate: IAnimate;
-}
+export type Width = 'small' | 'medium' | 'large' | string;
 
 export interface Props {
+  active?: boolean;
+  activator: React.ReactNode;
+  children?: any;
```

Comment:
@dheeraja00 We don't need children, React super class automatically gives us this.props.children.